### PR TITLE
prepare release 3.2.1

### PR DESCRIPTION
**Pull request by chaudum**
## Summary of the changes / Why this improves CrateDB

Prepare release notes and update version for release 3.2.1

## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -37,30 +37,14 @@ Unreleased Changes
 Breaking Changes
 ================
 
+None
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/unreleased.rst]: This is new to me, but I followed the instruction from here: https://github.com/crate/crate/pull/8094/files#diff-8918b36d12905ff85de79bcbdcf052aaR29

File: blackbox/docs/appendices/release-notes/3.2.1.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+.. NOTE::
+
+    If you are upgrading a cluster, you must be running CrateDB 2.0.4 or higher
+    before you upgrade to 3.2.1.
+
+    We recommend that you upgrade to the latest 3.1 release before moving to
+    3.2.1.
+
+    You cannot perform a `rolling upgrade`_ to this version. Any upgrade to this
```
Comment (seut) [inline blackbox/docs/appendices/release-notes/3.2.1.rst]: I think a rolling upgrade from 3.2.0 is possible.

File: blackbox/docs/appendices/release-notes/3.2.1.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Changelog
+=========
+
+Fixes
+-----
+
+- Fixed a ``NullPointerException`` that could occur using ``array_difference``.
+
+- Fixed a race condition that could lead to stuck queries. One case this could
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.2.1.rst]: is this the exhaustive list of cases? or just an example? I would like to suggest this bullet be rewritten but I don't know how to do that without knowing this

File: blackbox/docs/appendices/release-notes/3.2.1.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  occur is if a node was being stopped or crashed.
+
+- Fixed an issue that prevented queries on table functions from working if the
+  cluster contains an expired license.
+
+- Casts to nested arrays are now properly supported.
+
+- The type of parameter placeholders in sub-queries in the FROM clause of a
+  query can now be resolved to support postgresql clients relying on the
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.2.1.rst]: PostgreSQL

File: blackbox/docs/appendices/release-notes/3.2.1.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  ``ParameterDescription`` message.
+  This enables queries like ``select * from (select $1::int + $2) t``
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.2.1.rst]: add a new line above, or fold into para

**Pull request merged**